### PR TITLE
Pv2 2310 extend me timeout parameter

### PR DESCRIPTION
**Pull request by tom-gough**

### CODE REVIEW

File: src/SFA.DAS.Payments.JobContextMessageHandling/JobStatus/JobStatusService.cs [hunk 1]
```diff
@@ -35,11 +36,11 @@ public JobStatusService(IJobsDataContext dataContext, IPaymentLogger logger, IJo
             this.config = config ?? throw new ArgumentNullException(nameof(config));
         }
 
-        public async Task<bool> WaitForJobToFinish(long jobId, CancellationToken cancellationToken)
+        public async Task<bool> WaitForJobToFinish(long jobId, CancellationToken cancellationToken, bool isPeriodEndRun = false)
         {
             //TODO: Temp brittle solution to wait for jobs to finish
             logger.LogDebug($"Waiting for job {jobId} to finish.");
-            var endTime = DateTime.Now.Add(config.TimeToWaitForJobToComplete);
+            var endTime = isPeriodEndRun ? DateTime.Now.Add(config.TimeToWaitForPeriodEndRunJobToComplete) : DateTime.Now.Add(config.TimeToWaitForJobToComplete);
```
Comment (carlbm) [inline src/SFA.DAS.Payments.JobContextMessageHandling/JobStatus/JobStatusService.cs:43]: Being very picky here. Is it worth making this a method?

Comment (carlbm) [reply src/SFA.DAS.Payments.JobContextMessageHandling/JobStatus/JobStatusService.cs:43]: `JobRunTime(xxx)`

File: src/SFA.DAS.Payments.Monitoring.Jobs.Application/JobProcessing/JobStatusService.cs [hunk 2]
```diff
@@ -42,7 +42,7 @@ protected virtual Task<bool> CheckSavedJobStatus(JobModel job, CancellationToken
         protected virtual async Task<bool> IsJobTimedOut(JobModel job, CancellationToken cancellationToken)
         {
             var timedOutTime = DateTimeOffset.UtcNow;
-            if (job.Status != JobStatus.InProgress || job.StartTime.Add(Config.EarningsJobTimeout) >= timedOutTime)
+            if (job.Status != JobStatus.InProgress || job.StartTime.Add(job.JobType == JobType.PeriodEndRunJob ? Config.PeriodEndRunJobTimeout : Config.EarningsJobTimeout) >= timedOutTime)
```
Comment (carlbm) [inline src/SFA.DAS.Payments.Monitoring.Jobs.Application/JobProcessing/JobStatusService.cs:45]: Similar to the earlier comment, make this a method

**Review by carlbm — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.Payments.Monitoring.Jobs.Application/JobProcessing/JobStatusService.cs (hunk 2 above)
Comment (hardik-desai-kcl) [reply src/SFA.DAS.Payments.Monitoring.Jobs.Application/JobProcessing/JobStatusService.cs:45]: +1

File: src/SFA.DAS.Payments.JobContextMessageHandling/JobStatus/JobStatusService.cs (hunk 1 above)
Comment (tom-gough) [reply src/SFA.DAS.Payments.JobContextMessageHandling/JobStatus/JobStatusService.cs:43]: can do

**Review by carlbm — APPROVED** [review]

**Review by hardik-desai-kcl — APPROVED** [review]

**Pull request merged**